modules/phase_field/src/ics/SmoothCircleFromFileIC.C

```diff
@@ -0,0 +1,80 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "SmoothCircleFromFileIC.h"
+
+template <>
+InputParameters
+validParams<SmoothCircleFromFileIC>()
+{
+  InputParameters params = validParams<SmoothCircleBaseIC>();
+  params.addClassDescription("Multiple smooth circles read from a text file");
+  params.addRequiredParam<FileName>("file_name", "File containing circle centers and radii");
+
+  return params;
+}
+
+SmoothCircleFromFileIC::SmoothCircleFromFileIC(const InputParameters & parameters)
+  : SmoothCircleBaseIC(parameters),
+    _data(0),
+    _col_map(4),
+    _file_name(getParam<FileName>("file_name")),
+    _txt_reader(_file_name, true, " ", &_communicator),
+    _n_circles(0)
+{
+  // Read names and vectors from file
+  _txt_reader.read();
+  _col_names = _txt_reader.getColumnNames();
+  _data = _txt_reader.getColumnData();
+  _n_circles = _data[0].size();
+
+  // Check that the file has all the correct information
+  _col_map = {-1, -1, -1, -1};
+  for (unsigned int i = 0; i < _col_names.size(); ++i)
+  {
+    // Check that columns have uniform lengths
+    if (_data[i].size() != _n_circles)
+      mooseError("Columns in ", _file_name, " do not have uniform lengths.");
+
+    // Determine which columns correspond to which parameters.
+    if (_col_names[i] == "x")
+      _col_map[0] = i;
+    else if (_col_names[i] == "y")
+      _col_map[1] = i;
+    else if (_col_names[i] == "z")
+      _col_map[2] = i;
+    else if (_col_names[i] == "r")
+      _col_map[3] = i;
```

Comment:
Oh yes, I forgot that you can't use the `enum class` if you want to use the enum as an index. That syntax creates a strong type that has to be converted.

Instead drop the word "class" from your enum and then also drop the Prefix "COLS" when you try to use it. That should take care of your errors.